src/components/CharacterList.tsx

```diff
@@ -0,0 +1,44 @@
+import React, {useState} from 'react';
+import CharacterCard from './CharacterCard';
+import CharacterInfo from './CharacterInfo';
+
+const CharacterList = () => {
+  const [toggleDetails, setToggleDetails] = useState<boolean>(false);
+  return (
+    <div className='flex'>
+      <div
+        className={`mt-[8rem] ${
+          toggleDetails ? 'mr-[25rem]' : ''
+        } flex flex-wrap justify-between gap-10 w-full`}
+      >
+        <CharacterCard setToggleDetails={setToggleDetails} />
```

Comment:
pass character card data as props, 

set a variable that represents data that is fetched from the api, 

`
const data = [{
id: 1,
name: "bulbasaur",
type: [{id: 1, name"grass"}]
}]
`
Check the API response and set the variable accordingly so that you don't have to rename the variables declared in your components once you integrate the actual api